raster/r.watershed/ram/Gwater.h

```diff
@@ -131,7 +131,7 @@ int no_stream(int, int, CELL, double, CELL);
 int overland_cells(int, int, CELL, CELL, CELL *);
 
 /* ramseg.c */
-int size_array(int *, int, int);
+size_t size_array(int *, size_t, size_t);
```

Comment:
```suggestion
size_t size_array(int *, int, int);
```
in combination with local variables of size_t should be enough and perhaps better/cleaner. ncols and nrows comes from `Rast_window_rows()` in the form of integers anyway.

I tested with good results with:

```c
size_t size_array(int *ram_seg, int nrows_in, int ncols_in)
{
    size_t size, segs_in_col;
    size_t nrows = nrows_in;
    size_t ncols = ncols_in;
 ...
```

Reply:
This PR or your suggested change only fixes the memory allocation of certain arrays, but those arrays need to be indexed using an int type larger than signed `int` (e.g., `unsigned int` or `size_t`) because of their size (`nrows * ncols > INT32_MAX`) and some other arrays, function arguments, and local variables even contain those large indices. Figuring out which of them need to be promoted to a larger int type seems tricky...

Reply:
Right, I understood as much, just wanted go give a heads-up... :-)

All the same, there has been similar problems lately, fixed with #2853, #2844 and #2842.
There are also a number of related issues in https://github.com/OSGeo/grass/security/code-scanning under "Multiplication result converted to larger type".